### PR TITLE
Fixed - Elementor maintenance mode is not working

### LINKED ISSUE #386: Elementor maintenance mode is not working with BuddyBoss Platform
**Describe the bug**
Elementor maintenance mode is not working when BuddyBoss PLatform is enabled, this happens when a component page is set as homepage

**To Reproduce**
1. Go to Elementor > Tools
2. Under Maintenance Mode tab, set Choose Mode to Maintenance
3. Choose any template (or create if you don't have any)
4. Save changes
5. Go to home page and nothing was changed
6. Try to disable BuddyBoss Platform plugin and recheck the home page

**Expected behavior**
Homepage should be switched to maintenance mode when enabled in Elementor

**Screenshots**
mtnmode

**Screencast**
https://www.loom.com/share/1c6ff0ce41c6442f9b95f618f89daa89

**Support ticket links**
https://secure.helpscout.net/conversation/1112132586/14257/

### PR DESCRIPTION
**Pull request by asaquzzamanbb**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #386.

### How to test the changes in this Pull Request:

1. Go to Elementor > Tools.
2. Under Maintenance Mode tab, set Choose Mode to Maintenance.
3. Choose any template (or create if you don't have any).
4. Save changes.
5. Go to home page and nothing was changed.
6. Try to disable BuddyBoss Platform plugin and recheck the home page

### Proof Screenshots or Video

https://www.loom.com/share/c26a1d094d3b48f689b10fa4e9fbae52

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]
LGTM 🚀 

@asaquzzamanbb Merge in dev branch and add in testing.

**Review by mehulkaklotar — COMMENTED** [review]
@asaquzzamanbb Can we put this check in theme instead?

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 1]
```diff
@@ -527,3 +527,55 @@ function bp_core_set_uri_elementor_show_on_front( $bool ) {
...
 add_filter( 'bp_core_set_uri_show_on_front', 'bp_core_set_uri_elementor_show_on_front', 10, 3 );
+
+/**
+ * Prevent BB template redering and Redirect to the Elementor "Maintenance Mode" template.
+ *
+ * @since BuddyBoss 1.5.8
+ *
+ * @return void
+ */
+function bp_prevent_template_rendering() {
```
Comment (mehulkaklotar) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:538]: change function name to `bb_get_elementor_maintenance_mode_template`

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 2]
```diff
@@ -527,3 +527,55 @@ function bp_core_set_uri_elementor_show_on_front( $bool ) {
...
+		}
+	}
+
+	$mode = get_option( 'elementor_maintenance_mode_mode' );
+
+	if ( 'maintenance' === $mode || 'coming_soon' === $mode ) {
+		remove_action( 'template_redirect', 'bp_template_redirect', 10 );
+	}
+}
+add_action( 'bp_loaded', 'bp_prevent_template_rendering' );
```
Comment (mehulkaklotar) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:581]: function name change to `bb_get_elementor_maintenance_mode_template`

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]
@asaquzzamanbb please do the suggested changes.

**Review by mehulkaklotar — CHANGES_REQUESTED** [review]
@asaquzzamanbb please do the suggested changes.

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php (hunk 1 above)
Comment (asaquzzamanbb) [reply src/bp-core/compatibility/bp-incompatible-plugins-helper.php]: Update the function bp_prevent_template_rendering to bb_get_elementor_maintenance_mode_template.
https://github.com/buddyboss/buddyboss-platform/commit/b047fd6f00ea2b18a321e9d621667dad7e5af2a6

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php (hunk 2 above)
Comment (asaquzzamanbb) [reply src/bp-core/compatibility/bp-incompatible-plugins-helper.php]: Update the function bp_prevent_template_rendering to bb_get_elementor_maintenance_mode_template.
https://github.com/buddyboss/buddyboss-platform/commit/b047fd6f00ea2b18a321e9d621667dad7e5af2a6

**Review by chetansatasiya — APPROVED** [review]
@asaquzzamanbb Merge in dev and add in testing

**Comment by asaquzzamanbb** [comment]
@chetansatasiya Merged with dev branch.
https://github.com/buddyboss/buddyboss-platform/commit/e68ca3dd243b2e7edaae891006ebacbe1693cf1c

**Comment by sumitbuddyboss** [comment]
Hello @asaquzzamanbb 
Fixed > 
I have checked in the Dev branch and now maintenance mode is working properly with the platform for proof please check the below video.

Video
https://www.loom.com/share/df9a416935ea425bb0e9a8666a47d563


@chetansatasiya 
Ready for release 🚀

**Pull request merged**